### PR TITLE
Refactor Config Service

### LINKED ISSUE #6266: Blog title in admin panel as decimal number
### Issue Summary
Blog title in admin panel interpreted as a number, if a prefix is "0x"
### Steps to Reproduce
1. Change the name of the blog to "0xCD" (205 in decimal) for example
2. Refresh the page

### Screenshot
ghost

### PR DESCRIPTION
**Pull request by acburdine**
closes #6266
- add "type" to valid keys in configuration api
- refactor ember config service to parse values based on provided type

TODO:
- [x] fix tests

### CODE REVIEW

**Comment by acburdine** [comment]
@ErisDS I'm not sure if this is the best way to go about implementing the idea of "types" in the configuration, but it does fix the issue.

File: core/client/app/services/config.js [hunk 1]
```diff
@@ -2,27 +2,24 @@ import Ember from 'ember';
 
 const {Service, _ProxyMixin, computed} = Ember;
 
-function isNumeric(num) {
+function _isNumeric(num) {
```
Comment (halfdan) [inline core/client/app/services/config.js]: This is only used once and it's a one liner - just inline it where used?

Comment (kevinansfield) [inline core/client/app/services/config.js]: I'm :+1: on it, it's following the same semantics as the other `isBlank`, `isEmpty`, etc helper methods and makes it easily searchable if it's ever needed elsewhere.

@acburdine are you able to change the name from `_isNumeric` to `isNumeric` please? The convention for the `_` prefix is to indicate that a method or instance variable should be considered private, helper functions don't really fall into that category.

Comment (halfdan) [inline core/client/app/services/config.js]: @kevinansfield It literally just wraps the isNumeric from jQuery - so it's searchable. But :+1: for consistency with the other helpers.

Comment (kevinansfield) [inline core/client/app/services/config.js]: Agreed that it's already searchable but we've had a previous issue where the `isNumeric` method content was different and was changed to the current implementation which is why I feel the explicit naming is preferable - should probably have mentioned that in my last reply :smile:

**Comment by acburdine** [comment]
@kevinansfield fixed :smile:

**Comment by kevinansfield** [comment]
@ErisDS I'm happy with the client-side changes here.

File: core/client/app/services/config.js [hunk 2]
```diff
@@ -6,23 +6,20 @@ function isNumeric(num) {
...
     if (val === '') {
         return null;
-    } else if (val === 'true') {
-        return true;
-    } else if (val === 'false') {
-        return false;
-    } else if (isNumeric(val)) {
+    } else if (type === 'bool') {
+        return (val === 'true') ? true : false;
+    } else if (type === 'int' && _isNumeric(val)) {
```
Comment (kevinansfield) [inline core/client/app/services/config.js]: I think you missed the rename here :smile:

**Pull request merged**